setpass/api.py

```diff
@@ -176,13 +180,17 @@ def view_reset_form():
 @wsgi.app.route('/reset', methods=['POST'])
 def reset_password():
     name = request.form['name']
-    username = request.form['username']
+    email = request.form['email']
+    confirm_email = request.form['confirm_email']
     pin = request.form['pin']
 
-    if not name or not username or not pin:
-        return Response(response='Missing name/pin/username!', status=400)
+    if not name or not email or not confirm_email or not pin:
+        return Response(response='Missing required field!', status=400)
+
+    if email != confirm_email:
+        return Response(response="Email addresses do not match.", status=400)
```

Comment:
I'm not an expert or pedant regarding HTTP status code semantics, but the 'do not match' case seems to be inconsistently 400 or 403. (compare this line, 191, with 57)